sled-agent/zone-images/src/test_utils.rs

```diff
@@ -0,0 +1,343 @@
+    LazyLock::new(|| OverridePaths::for_uuid(BOOT_UUID));
+
+pub(crate) const NON_BOOT_UUID: ZpoolUuid =
+    ZpoolUuid::from_u128(0x4854189f_b290_47cd_b076_374d0e1748ec);
+pub(crate) const NON_BOOT_ZPOOL: ZpoolName =
+    ZpoolName::new_internal(NON_BOOT_UUID);
+pub(crate) static NON_BOOT_PATHS: LazyLock<OverridePaths> =
+    LazyLock::new(|| OverridePaths::for_uuid(NON_BOOT_UUID));
+
+pub(crate) const NON_BOOT_2_UUID: ZpoolUuid =
+    ZpoolUuid::from_u128(0x72201e1e_9fee_4231_81cd_4e2d514cb632);
+pub(crate) const NON_BOOT_2_ZPOOL: ZpoolName =
+    ZpoolName::new_internal(NON_BOOT_2_UUID);
+pub(crate) static NON_BOOT_2_PATHS: LazyLock<OverridePaths> =
+    LazyLock::new(|| OverridePaths::for_uuid(NON_BOOT_2_UUID));
+
+pub(crate) const NON_BOOT_3_UUID: ZpoolUuid =
+    ZpoolUuid::from_u128(0xd0d04947_93c5_40fd_97ab_4648b8cc28d6);
+pub(crate) const NON_BOOT_3_ZPOOL: ZpoolName =
+    ZpoolName::new_internal(NON_BOOT_3_UUID);
+pub(crate) static NON_BOOT_3_PATHS: LazyLock<OverridePaths> =
+    LazyLock::new(|| OverridePaths::for_uuid(NON_BOOT_3_UUID));
+
+pub(crate) fn make_internal_disks_rx(
+    root: &Utf8Path,
+    boot_zpool: ZpoolName,
+    other_zpools: &[ZpoolName],
+) -> InternalDisksReceiver {
+    let identity_from_zpool = |zpool: ZpoolName| DiskIdentity {
+        vendor: "sled-agent-zone-images-tests".to_string(),
+        model: "fake-disk".to_string(),
+        serial: zpool.id().to_string(),
+    };
+    let mount_config = MountConfig {
+        root: root.to_path_buf(),
+        synthetic_disk_root: root.to_path_buf(),
+    };
+    InternalDisksReceiver::fake_static(
+        Arc::new(mount_config),
+        std::iter::once((identity_from_zpool(boot_zpool), boot_zpool)).chain(
+            other_zpools
+                .iter()
+                .copied()
+                .map(|pool| (identity_from_zpool(pool), pool)),
+        ),
+    )
+}
+
+/// Context for writing out fake zones to install dataset directories.
+///
+/// The tests in this module ensure that the override JSON's list of zones
+/// matches the zone files on disk.
+#[derive(Clone, Debug)]
+pub(crate) struct WriteInstallDatasetContext {
+    pub(crate) zones: IdOrdMap<ZoneContents>,
+    pub(crate) mupdate_id: MupdateUuid,
+    pub(crate) mupdate_override_uuid: MupdateOverrideUuid,
+    write_zone_manifest_to_disk: bool,
+}
+
+impl WriteInstallDatasetContext {
+    /// Initializes a new context with a couple of zones and no known
+    /// errors.
+    pub(crate) fn new_basic() -> Self {
+        Self {
+            zones: [
+                ZoneContents::new("zone1.tar.gz", b"zone1"),
+                ZoneContents::new("zone2.tar.gz", b"zone2"),
+                ZoneContents::new("zone3.tar.gz", b"zone3"),
+                ZoneContents::new("zone4.tar.gz", b"zone4"),
+                ZoneContents::new("zone5.tar.gz", b"zone5"),
+            ]
+            .into_iter()
+            .collect(),
+            mupdate_id: MupdateUuid::new_v4(),
```

Comment:
I think this is causing the new test failures, maybe? The display impl now includes this ID, which is different each time the test runs?

Reply:
This is what I get for writing code at midnight. Thanks.